PWGHF/D2H/TableProducer/dataCreatorDplusPiReduced.cxx

```diff
@@ -188,38 +207,31 @@ struct HfDataCreatorDplusPiReduced {
     return true;
   }
 
-  void process(aod::Collisions const& collisions,
-               CandsDFiltered const& candsD,
-               aod::TrackAssoc const& trackIndices,
-               TracksPIDWithSel const&,
-               aod::BCsWithTimestamps const&)
+  template <bool doMc = false, typename P, typename T>
+  void runDataCreation(aod::Collisions const& collisions,
+                       CandsDFiltered const& candsD,
+                       aod::TrackAssoc const& trackIndices,
+                       T const& tracks,
+                       P const& particlesMc,
+                       aod::BCsWithTimestamps const& bcs)
   {
     // store configurables needed for B0 workflow
     if (!isHfCandB0ConfigFilled) {
-      rowCandidateConfig(selectionFlagD.value);
+      rowCandidateConfig(selectionFlagD.value, invMassWindowDPi.value);
       isHfCandB0ConfigFilled = true;
     }
 
-    static int aodNumber = 0;
-    aodNumber++;
-
     // handle normalization by the right number of collisions
     hfCollisionCounter(collisions.tableSize());
 
-    static int nCol = 0;
     for (const auto& collision : collisions) {
-      nCol++;
 
       // helpers for ReducedTables filling
-      int hfReducedCollisionIndex = hfReducedCollision.lastIndex() + 1;
-      std::vector<int64_t> selectedTracksPion;
-      selectedTracksPion.reserve(trackIndices.size());
+      int indexHfReducedCollision = hfReducedCollision.lastIndex() + 1;
+      std::map<int64_t, int64_t> selectedTracksPion;
```

Comment:
For better code readability, here I'd add a comment explaining that:
 - the `key` is the `track.globalIndex()`
 - the `value` is the track index in the table of the found HF pions